packages/console/src/components/ReportConversion/index.tsx

```diff
@@ -1,20 +1,47 @@
 import { useEffect } from 'react';
 import { Helmet } from 'react-helmet';
 
+import useCurrentUser from '@/hooks/use-current-user';
+
 import {
   shouldReport,
   lintrk,
   gtagAwTrackingId,
   linkedInConversionId,
   gtag,
   gtagSignUpConversionId,
+  rdt,
+  redditAdAccountId,
+  hashEmail,
 } from './utils';
 
 /**
  * In cloud production environment, this component initiates gtag.js and LinkedIn
  * Insight Tag, then reports a sign-up conversion to them.
  */
 export default function ReportConversion() {
+  const { user, isLoading } = useCurrentUser();
+
+  /**
+   * Initiate Reddit Pixel and report a sign-up conversion to it when user is loaded.
+   * Use user email to prevent duplicate conversion, and it is hashed before sending
+   * to protect user privacy.
+   */
+  useEffect(() => {
+    const report = async () => {
+      rdt('init', redditAdAccountId, {
+        optOut: false,
+        useDecimalCurrencyValues: true,
+        email: await hashEmail(user?.primaryEmail ?? undefined),
```

Comment:
nit: Is the fallback to undefined necessary? I see the hashEmail function already taken care of falsy values inside.

Reply:
`primaryEmail` can be `null`. it's ok in runtime, here for type rigorousness